app/services/store.js

```diff
@@ -43,21 +43,25 @@ export default class ExtendedStoreService extends Store {
       firstRelationship
     );
     const isMany = firstRelInfo.kind === "hasMany";
+    const firstRelationshipData = (await this.query(modelInstance.get('constructor.modelName'), {
```

Comment:
I think maybe a cleaner way is to change the signature of this method and work with the id of the modelInstance directly
that means we can use a `findRecord` to get the instance, and then go from there
I think there's no way around the fact that we have to fetch the object (again), but if we can we should let ember-data decide how, it might do some clever caching? or something? who knows 

In any case it will remove the illusion that this method does not refetch the modelinstance

Reply:
Not sure I understand what would be different between findRecord and query, and if we get only the record id I think we cannot get the modelName, we should also ask for that right?

Reply:
iirc ember data can be smarter about findRecord
with query it will always make the request 
but tbh it probably doesnt matter

and yes, youd need the modelname as well

Reply:
so in short: I think its better to just move to asking for modelname and id, along with the relationship path
that way we are sure that we have the right instance and its more clear to the caller we will make a full backend request 

@nvdk any thoughts?

Reply:
I understand the question is: what makes the most sense as a method signature?

In that case I think passing modelname and id, instead of the modelinstance is indeed more clear as to what will actually happen.

Reply:
yeah
whether we use findrecord or query then shouldn't really matter here I think

Reply:
@lagartoverde  will you take this up?

Reply:
Yes :), probably won't finish today as I have to change every call to this method

Reply:
I don't think it's used very often